cloudflare/resource_cloudflare_teams_route.go

```diff
@@ -0,0 +1,134 @@
+package cloudflare
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareTeamsRoute() *schema.Resource {
+	return &schema.Resource{
+		Schema: resourceCloudflareTeamsRouteSchema(),
+		Create: resourceCloudflareTeamsRouteCreate,
+		Read:   resourceCloudflareTeamsRouteRead,
+		Update: resourceCloudflareTeamsRouteUpdate,
+		Delete: resourceCloudflareTeamsRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareTeamsRouteImport,
+		},
+	}
+}
+
+func resourceCloudflareTeamsRouteRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	tunnelRoute, err := client.GetTunnelRouteForIP(context.Background(), cloudflare.TunnelRoutesForIPParams{
```

Comment:
The use of this to read the resource seems fragile.  If you are managing two routes, say `10.0.0.0/8` and `10.10.0.0/24` via terraform and `10.10.0.0/24` gets manually removed on accident ... terraform will scan resources and actually find the route for `10.0.0.0/8` for the `10.10.0.0/24` resource.  Was it not feasible to use the List Tunnel Routes API passing in the tunnel_id parameter to specify an exact tunnel?

Reply:
Alternatively, perhaps this should verify the network returned is an exact match for what was requested to prevent this?

Reply:
i guess more than one network could route to the same tunnel, but you could pass in `network_subnet=10.10.0.0/24` and `nework_superset=10.10.0.0/24` to get the exact route.  And probably include `is_deleted=false` too ...

Reply:
I'll take a look at doing that instead. It's a lot of workaround for an endpoint that doesn't follow the normal pattern of the  V4 API.